test/RequestStreamTest.cpp

```diff
@@ -120,3 +145,29 @@ TEST(RequestStreamTest, RequestOnDisconnectedClient) {
   CHECK_WAIT(wait_for_on_error);
   ASSERT(did_call_on_error);
 }
+
+class TestHandlerError : public rsocket::RSocketResponder {
+ public:
+  Reference<Flowable<Payload>> handleRequestStream(
+      Payload request,
+      StreamId) override {
+    // string from payload data
+    auto requestString = request.moveDataToString();
+
+    return Flowables::error<Payload>(folly::exception_wrapper(
+        std::runtime_error("A wild error appeared")));
+  }
+};
+
+TEST(RequestStreamTest, HandleError) {
+  folly::ScopedEventBaseThread worker;
+  auto server = makeServer(std::make_shared<TestHandlerError>());
+  auto client = makeClient(worker.getEventBase(), *server->listeningPort());
+  auto requester = client->getRequester();
+  auto ts = TestSubscriber<std::string>::create();
+  requester->requestStream(Payload("Bob"))
+           ->map([](auto p) { return p.moveDataToString(); })
+           ->subscribe(ts);
+  ts->awaitTerminalEvent();
+  ts->assertOnErrorMessage("A wild error appeared");
```

Comment:
Add a test for what happens when we receive {0, 1, many} elements and then an error
Also test what happens when an exception is thrown in:
 - thrown in handleRequestStream
 - return a Flowable::error
 -  return a Flowable::create which emits elements, and within the Flowable throws 

and assertValueCount for those cases?

Reply:
* thrown in handleRequestStream
_Unless caught, and something (Flowables::error probably) is sent, what would we want to happen?  An uncaught throw should crash no?_
* return a Flowable::error
_Isn't this what testHandlerError is doing?_
* return a Flowable::create which emits elements, and within the Flowable throws
_Same story as above, if we're catching it, we're just sending a Flowables::error, and if not, it should crash no?_

I guess I could kind of see doing this and confirming that the correct number of elements made it through before the crash?  That's not really testing any path that we're not seeing in the normal HelloSync test.  I'm not seeing how this adds to the test cases.  Am I misunderstanding what you're suggesting?

Reply:
Threw together an example that throws after a few go out, and it just crashes the whole thing.  

```
class TestErrorThrown : public rsocket::RSocketResponder {
 public:
  Reference<Flowable<Payload>> handleRequestStream(Payload request, StreamId)
      override {
    // string from payload data
    auto requestString = request.moveDataToString();
      return Flowables::range(1, 10)->map([name = std::move(requestString)](
          int64_t v) {
        if (v == 5) {
          throw std::runtime_error{"What an unexpected turn of events!"};
        }
        std::stringstream ss;
        ss << "Hello " << name << " " << v << "!";
        std::string s = ss.str();
        return Payload(s, "metadata");
      });
  }
};
```

Do we expect better behavior?